### DISCUSSION TITLE
Installation API to run Installation Wizard programmatically

### DISCUSSION THREAD

**Discussion by sriwantha** [dnnsoftware/Dnn.Platform#5612] [Q&A]
Is there a way I can run the Installation wizard /Install/InstallWizard.aspx without running in manually? E.g. command line or api I can invoke to run the installation wizard with pre-configured set of values. For example, is there a web-api I can query to run the installation instead of running it manually. https://www.dnnsoftware.com/docs/administrators/setup/run-installation-wizard.html

Situation: I have to programetically provision DNN with a fixed template, database and language to a given MS SQL server instance. The installation should happen as part of the devops pipeline.

**Comment by sleupold** [comment]
AFAIK, this doesn't work with the Installwizard, but you cah use install.aspx?mode=install, and it will use values already entered into the web.config and default portal template.

**Reply by sriwantha** [reply]
How do you provide the default website admin username and password in this situation. What should I include in the web.config?

**Reply by valadas** [reply]
I _think_ you could customize /install/DotNetNuke.install.config.resources before calling the installer. I have not tried it though, so....

**Comment by sleupold** [comment]
if you use install.aspx, it will use default credentials ("host" > "dnnhost"), you need to customize directly after installation
